internal/service/emr/studio.go

```diff
@@ -0,0 +1,274 @@
+	}
+
+	d.SetId(aws.StringValue(result.StudioId))
+
+	return resourceStudioRead(d, meta)
+}
+
+func resourceStudioUpdate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*conns.AWSClient).EMRConn
+
+	if d.HasChangesExcept("tags", "tags_all") {
+		input := &emr.UpdateStudioInput{
+			StudioId: aws.String(d.Id()),
+		}
+
+		if d.HasChange("description") {
+			input.Description = aws.String(d.Get("description").(string))
+		}
+
+		if d.HasChange("name") {
+			input.Name = aws.String(d.Get("name").(string))
+		}
+
+		if d.HasChange("default_s3_location") {
+			input.DefaultS3Location = aws.String(d.Get("default_s3_location").(string))
+		}
+
+		if d.HasChange("subnet_ids") {
+			input.SubnetIds = flex.ExpandStringSet(d.Get("subnet_ids").(*schema.Set))
+		}
+	}
+
+	if d.HasChange("tags_all") {
+		o, n := d.GetChange("tags_all")
+
+		if err := UpdateTags(conn, d.Id(), o, n); err != nil {
+			return fmt.Errorf("error updating EMR Studio (%s) tags: %w", d.Id(), err)
+		}
+	}
+
+	return resourceStudioRead(d, meta)
+}
+
+func resourceStudioRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*conns.AWSClient).EMRConn
+	defaultTagsConfig := meta.(*conns.AWSClient).DefaultTagsConfig
+	ignoreTagsConfig := meta.(*conns.AWSClient).IgnoreTagsConfig
+
+	studio, err := FindStudioByID(conn, d.Id())
+	if !d.IsNewResource() && tfresource.NotFound(err) {
+		log.Printf("[WARN] EMR Studio (%s) not found, removing from state", d.Id())
+		d.SetId("")
+		return nil
+	}
+
+	if err != nil {
+		return fmt.Errorf("error reading EMR Studio (%s): %w", d.Id(), err)
+	}
+
+	d.Set("arn", studio.StudioArn)
+	d.Set("auth_mode", studio.AuthMode)
+	d.Set("default_s3_location", studio.DefaultS3Location)
+	d.Set("description", studio.Description)
+	d.Set("engine_security_group_id", studio.EngineSecurityGroupId)
+	d.Set("idp_auth_url", studio.IdpAuthUrl)
+	d.Set("idp_relay_state_parameter_name", studio.IdpRelayStateParameterName)
+	d.Set("name", studio.Name)
+	d.Set("service_role", studio.ServiceRole)
+	d.Set("url", studio.Url)
+	d.Set("user_role", studio.UserRole)
+	d.Set("vpc_id", studio.VpcId)
+	d.Set("workspace_security_group_id", studio.WorkspaceSecurityGroupId)
+	d.Set("subnet_ids", flex.FlattenStringSet(studio.SubnetIds))
+
+	tags := KeyValueTags(studio.Tags).IgnoreAWS().IgnoreConfig(ignoreTagsConfig)
+
+	//lintignore:AWSR002
+	if err := d.Set("tags", tags.RemoveDefaultConfig(defaultTagsConfig).Map()); err != nil {
+		return fmt.Errorf("error setting tags: %w", err)
+	}
+
+	if err := d.Set("tags_all", tags.Map()); err != nil {
+		return fmt.Errorf("error setting tags_all: %w", err)
+	}
+
+	return nil
+}
+
+func resourceStudioDelete(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*conns.AWSClient).EMRConn
+
+	request := &emr.DeleteStudioInput{
+		StudioId: aws.String(d.Id()),
+	}
+
+	log.Printf("[INFO] Deleting EMR Studio: %s", d.Id())
+	_, err := conn.DeleteStudio(request)
```

Comment:
Should there be a waiter here?

Reply:
doesnt look like studio exposes any status attribute so no, also the post destroy checks pass